### PR TITLE
Make ejecting from devices configurable in ball search.

### PR DESCRIPTION
**Pull request by dziedada**
Added functionality and test.
The test case adapts a previous test, test_ball_search_iterations_and_give_up_with_new_ball, but changes the config file, adds relevant asserts, and removes irrelevant asserts and simplifies the advancing of time and running in the test.

### CODE REVIEW

File: mpf/tests/machine_files/ball_search/config/no_eject.yaml [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+        tags: trough, drain, home
+        ball_search_order: 1
+    test_launcher:
+        eject_coil: eject_coil2
+        ball_switches: s_ball_switch_launcher
+        eject_timeouts: 5s
+        eject_coil_jam_pulse: 5ms
+        debug: true
+        ball_search_order: 2
+        tags: no-eject
```
Comment (jabdoa2) [inline mpf/tests/machine_files/ball_search/config/no_eject.yaml]: What do you think about making the tag a bit more self-explanatory? Something like "no-eject-on-ballsearch"

File: mpf/devices/ball_device/pulse_coil_ejector.py [hunk 2]
```diff
@@ -73,11 +73,13 @@ def ball_search(self, phase, iteration):
                 return self._fire_coil_for_search(True)
         elif phase == 2:
             # round 2: all devices except trough. small pulse
-            if 'trough' not in self.ball_device.config['tags']:
+            if 'trough' not in self.ball_device.config['tags'] and \
+                    'no-eject' not in self.ball_device.config['tags']:
                 return self._fire_coil_for_search(False)
```
Comment (jabdoa2) [inline mpf/devices/ball_device/pulse_coil_ejector.py]: Looks good! Maybe we can still pulse when idle + zero balls? Alternatively we should also disable the first pulse. Just to keep it consistent.

**Review by jabdoa2 — COMMENTED** [review]
Looks very good. Just some small comments.

File: mpf/devices/ball_device/pulse_coil_ejector.py (hunk 2 above)
Comment (dziedada) [reply mpf/devices/ball_device/pulse_coil_ejector.py]: For all phases of ball search then?

File: mpf/tests/machine_files/ball_search/config/no_eject.yaml (hunk 1 above)
Comment (dziedada) [reply mpf/tests/machine_files/ball_search/config/no_eject.yaml]: Sounds good, I'll change it.

File: mpf/devices/ball_device/pulse_coil_ejector.py (hunk 2 above)
Comment (dziedada) [reply mpf/devices/ball_device/pulse_coil_ejector.py]: Took the second alternative and disabled the first pulse to keep it consistent.

**Comment by jabdoa2** [comment]
fixes #1204

**Comment by jabdoa2** [comment]
Perfect. Thanks!

**Pull request merged**